int32 sid
---
bool was_set
